File: kenenisa/pascals-triangle.py
module Triangle (rows) where

row :: [Int] -> [Int]
row [] = []
row [x] = [x]
row (x:y:xs) = x+y : row (y:xs)

rows :: Int -> [[Int]]
rows 0 = []
rows 1 = [[1]]
rows 2 = [[1],[1,1]]
rows x = prevRow ++ [1 : row (last prevRow)]
        where prevRow = rows (x - 1)

File: leul/exercism/PascalsTriangle.hs
-- https://exercism.org/tracks/haskell/exercises/pascals-triangle/

module Triangle (rows) where


addTwo :: [Integer] -> [Integer]
addTwo [] = []
addTwo [x] = []
addTwo (x:xs) = (x + head xs) : addTwo xs


row :: Int -> [Integer]
row 1 = [1]
row n = [1] ++ addTwo (row (n-1)) ++ [1]

rows :: Int -> [[Integer]]
rows = reverse . buildRows 
    where buildRows 0 = []
          buildRows x = row x : buildRows (x - 1)


